zcash_primitives/src/sapling.rs

```diff
@@ -185,15 +185,36 @@ impl ProofGenerationKey {
     pub fn to_viewing_key(&self) -> ViewingKey {
         ViewingKey {
             ak: self.ak,
-            nk: constants::PROOF_GENERATION_KEY_GENERATOR * self.nsk,
+            nk: NullifierDerivingKey(constants::PROOF_GENERATION_KEY_GENERATOR * self.nsk),
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct NullifierDerivingKey(jubjub::SubgroupPoint);
+
+impl From<jubjub::SubgroupPoint> for NullifierDerivingKey {
+    fn from(p: jubjub::SubgroupPoint) -> Self {
+        NullifierDerivingKey(p)
+    }
+}
+
+impl From<NullifierDerivingKey> for jubjub::SubgroupPoint {
+    fn from(nk: NullifierDerivingKey) -> Self {
+        nk.0
+    }
+}
+
+impl NullifierDerivingKey {
+    pub fn to_bytes(&self) -> [u8; 32] {
```

Comment:
Documentation. Also should this be public? Are we likely to want to serialize this independently of doing so inside other viewing key serializations? (Presumably no given that we can't do anything with the encoding, as no `from_bytes` method is present.)

Reply:
I'll drop this back out. I initially added it with the idea that it helped a bit with encapsulation, but you're right that it's only really relevant inside of ZIP 32.